app/models/application.rb

```diff
@@ -10,14 +10,43 @@ class Application < ApplicationRecord
   has_many :authentications, :through => :application_authentications
 
   attribute :availability_status, :string
-  validates :availability_status, :inclusion => { :in => %w[available unavailable] }, :allow_nil => true
+  validates :availability_status, :inclusion => {:in => %w[available unavailable]}, :allow_nil => true
 
   validate :source_must_be_compatible
 
+  before_save :copy_superkey_data
+  after_create :superkey_workflow
+  after_destroy :superkey_workflow
+
   private
 
   def source_must_be_compatible
-  	return if application_type.supported_source_types.include?(source.source_type.name)
-  	errors.add(:source, "of type: #{source.source_type.name}, is not compatible with this application type")
+    return if application_type.supported_source_types.include?(source.source_type.name)
+
+    errors.add(:source, "of type: #{source.source_type.name}, is not compatible with this application type")
+  end
+
+  def superkey_workflow
+    return unless source.app_creation_workflow == "account_authorization"
```

Comment:
if the string will be used on more places, I vote for constants :)

Reply:
I think this might be the only place but I constantized it anyway.